### PR TITLE
JS: Improve JSX lexing (support fragment syntax and embedded comments)

### PR DESCRIPTION
**Pull request by matthiasblaesing**
There are two issues in the JSX lexers:

- embedded comments in assignment blocks are not correctly handled
- fragment syntax (tag without name `<></>` was not supported

To visualize (on the left is the before picture, on the right is the after picture):

image

Closes: #5149
Closes: #4832

### CODE REVIEW

File: webcommon/libs.nashorn/src/com/oracle/js/parser/Parser.java [hunk 1]
```diff
@@ -5958,6 +5971,9 @@ private Expression jsxAttribute() {
     }
 
     private String jsxElementName() {
+        if(type == TokenType.JSX_ELEM_END) {
```
Comment (junichi11) [inline webcommon/libs.nashorn/src/com/oracle/js/parser/Parser.java:5974]: Nitpick: formatting

**Review by junichi11 — APPROVED** [review]
I've verified that there is no error with the example code.
nb-gh-5765-1
nb-gh-5765-2

**Comment by matthiasblaesing** [comment]
@junichi11 thank you for the check. I'm updating this PR in place with the fix for the formatting and will merge once it is green.

I removed the `ci:dev-build` label as a new build is not necessary (unittests will be run).

**Comment by locness3** [comment]
Thank you very much!

**Pull request merged**